File: Backend/TrackrAPI/TrackrAPI/Services/Seguridad/RolService.cs
using TrackrAPI.Dtos.Catalogo;
using TrackrAPI.Dtos.Seguridad;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Catalogo;
using TrackrAPI.Repositorys.Seguridad;
using System.Collections.Generic;


namespace TrackrAPI.Services.Seguridad
{
    public class RolService
    {

        private IRolRepository rolRepository;
        private RolValidatorService rolValidatorService;

        public RolService(IRolRepository rolRepository,
            RolValidatorService rolValidatorService)
        {
            this.rolRepository = rolRepository;
            this.rolValidatorService = rolValidatorService;

        }

        public Rol Consultar(int idRol)
        {
            return rolRepository.Consultar(idRol);
        }

        public RolDto ConsultarDto(int idRol)
        {
            var rol = rolRepository.ConsultarDto(idRol);
            rolValidatorService.ValidarExistencia(rol);
            return rol;
        }

        public IEnumerable<RolGridDto> ConsultarTodosParaGrid(int idCompania)
        {
            return rolRepository.ConsultarTodosParaGrid(idCompania);
        }

        public IEnumerable<RolDto> ConsultarTodosParaSelector()
        {
            return rolRepository.ConsultarTodosParaSelector();
        }

        public IEnumerable<RolDto> ConsultarPorUsuario(int idUsuario)
        {
            return rolRepository.ConsultarPorUsuario(idUsuario);
        }

        public void Agregar(Rol rol)
        {
            rolValidatorService.ValidarAgregar(rol);
            rolRepository.Agregar(rol);
        }

        public void Editar(Rol rol)
        {
            rolValidatorService.ValidarEditar(rol);
            rolRepository.Editar(rol);
        }

        public void Eliminar(int idRol)
        {
            var rol = rolRepository.Consultar(idRol);
            rolValidatorService.ValidarEliminar(idRol);
            rolRepository.Eliminar(rol);
        }

    }
}

File: Frontend/trackr-administrador/src/app/shared/http/seguridad/rol.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Rol } from '@models/seguridad/rol';

@Injectable()
export class RolService {
  private dataUrl = 'rol/'

  constructor(private http: HttpClient) { }

  public consultarTodosParaSelector() {
    return this.http.get<Rol[]>(this.dataUrl + 'consultarTodosParaSelector');
  }

  public consultarTodosParaGrid() {
    return this.http.get<Rol[]>(this.dataUrl + 'consultarTodosParaGrid');
  }

  public consultarRolesUsuarioSesion() {
    return this.http.get<Rol[]>(this.dataUrl + 'consultarRolesUsuarioSesion');
  }

  public consultar(idRol: number) {
    return this.http.get<Rol>(this.dataUrl + `consultar/${idRol}`);
  }

  public agregar(rol: Rol) {
    return this.http.post<void>(this.dataUrl + 'agregar', rol);
  }

  public editar(rol: Rol) {
    return this.http.put<void>(this.dataUrl + 'editar', rol);
  }

  public eliminar(idRol: number) {
    return this.http.delete<void>(this.dataUrl + `eliminar/${idRol}`);
  }
}
